### PR TITLE
Update API errors handling

### PR DESCRIPTION
**Pull request by yurii-ve**
Fixes https://github.com/wixplosives/codux/issues/22381

- Changed `EcomAPI` methods to return `EcomAPIResponse` type, which simplifies error handling on the client
```
type EcomAPISuccessResponse<T> = { status: 'success'; body: T };
type EcomAPIFailureResponse = { status: 'failure'; error: { code: EcomApiErrorCodes; message?: string } };
type EcomAPIResponse<T> = EcomAPISuccessResponse<T> | EcomAPIFailureResponse;
```

- Added ErrorBoundary to the category page

- Increase prettier  "printWidth" from 100 to 120

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -1,9 +1,18 @@
 {
-  "printWidth": 100,
+  "printWidth": 120,
```
Comment (VladyslavG) [inline .prettierrc:2]: this isn't mentioned in PR description. is this accidental ?

File: src/api/api-hooks.ts [hunk 2]
```diff
@@ -36,10 +50,7 @@ export const useAddToCart = () => {
             }
             const itemInCart = findItemIdInCart(cart, arg.id, arg.options);
             return itemInCart
-                ? ecomApi.updateCartItemQuantity(
-                      itemInCart._id,
-                      (itemInCart.quantity || 0) + arg.quantity
-                  )
+                ? ecomApi.updateCartItemQuantity(itemInCart._id, (itemInCart.quantity || 0) + arg.quantity)
```
Comment (VladyslavG) [inline src/api/api-hooks.ts]: `itemInCart.quantity ||` -> `??`

File: src/utils/index.ts [hunk 3]
```diff
@@ -2,3 +2,26 @@ export function getUrlOriginWithPath(url: string) {
...
+export function toError(value: unknown): Error {
+    if (value instanceof Error) {
+        return value;
+    }
+
+    if (typeof value === 'undefined') {
+        return new Error();
+    }
+
+    let errorMessage = String(value);
```
Comment (VladyslavG) [inline src/utils/index.ts]: it is weird to unconditionally convert potential object into string, if we can avoid it

```
export function toError(value: unknown): Error {
    if (value instanceof Error) {
        return value;
    }

    if (typeof value === 'undefined') {
        return new Error();
    }

    if (typeof value === 'object' && value !== null) {
        if ('message' in value) return new Error(String(value.message));

        if ('data' in value) {
            return new Error(String(value.data));
        }
    }

    return new Error(String(value));
}
```

**Review by olehrakwix — APPROVED** [review]

File: .prettierrc (hunk 1 above)
Comment (yurii-ve) [reply .prettierrc:2]: not accidental.. updated PR description

File: src/utils/index.ts (hunk 3 above)
Comment (yurii-ve) [reply src/utils/index.ts]: @olehrakwix please review slightly refactored `toError` util

**Review by VladyslavG — APPROVED** [review]

**Pull request merged**